### PR TITLE
[historical] throw error if given a string for period{1,2} that is invalid input for new Date()

### LINKED ISSUE #457: [historical] throw error if given a string for period{1,2} that is invalid input for new Date()
```bash
$ yahoo-finance historical TSLA '{"period1":"2022-02-01","period2":"today"}'
https://query2.finance.yahoo.com/v7/finance/download/TSLA?interval=1d&events=history&includeAdjustedClose=true&period1=1643673600&period2=NaN
Exiting with HTTPError: Bad Request
```

We should validate `period1` and `period2` and throw an error on invalid input before trying to make the request.

As reported by @kiloppertry in #429.

### PR DESCRIPTION
**Pull request by glaucoheitor**
Closes #457  .

## Changes
- Fixes #457
- Fixes small typo in CONTRIBUTING.md

## Type
- [ ] New Module
- [ ] Other New Feature
- [ ] Validation Fix
- [X] Other Bugfix
- [X] Docs
- [ ] Chore/other

## Comments/notes

First time submitting a 'real' pull request with code. I didn't add tests because I'm not familiar with it.
Also, I'm not 100% sure if my change is the best decision for this issue, but I try a few different options and I liked this the most. Please let me know if I can improve this further and if I made something wrong so I can improve next time. Thanks for the amazing project, as always! :)

### CODE REVIEW

**Comment by gadicc** [comment]
Hey @glaucoheitor, awesome work!  Thanks for stepping up :raised_hands:  It's really great for maintainers when users are contributing back and this is greatly appreciated :grin:

Yes, I was also thinking of using `isNaN`, so we're on the same page.  And throwing an error exactly as you have.

I have two minor comments which I'll place inline.  And no problem re the test, I'll do that after we merge and I think it will make a lot of sense to you once you see it :D

Thanks again!

File: src/modules/historical.ts
Comment (gadicc) [inline src/modules/historical.ts:76]: `timestamp <= 0` is actually ok :)  because 0 is the unix epoch of Jan 1st, 1970 GMT.  So we use negative numbers for dates before that (try e.g. `new Date(-3_600_000) // -1hr` and see what pops up).  I don't know how far back Yahoo's data goes, but let's leave it up to the user to play with that :D  So let's keep just the `isNaN` check,

File: src/modules/historical.ts
Comment (gadicc) [inline src/modules/historical.ts:81]: This is great but we can help the user even more by letting them know what the troublesome input was, i.e. `"' invalid date provided: '" + value + "'"` or something.

**Review by gadicc — CHANGES_REQUESTED** [review]

File: src/modules/historical.ts
Comment (glaucoheitor) [reply src/modules/historical.ts]: You're right, that makes total sense. I updated the commit.

**Comment by glaucoheitor** [comment]
I'm glad to help @gadicc. I really like this project and makes me happy to be able to help just a lil bit. Thanks for the opportunity and your patience.

**Review by gadicc — APPROVED** [review]

**Comment by gadicc** [comment]
Awesome, thanks @glaucoheitor!  For sure, and all help is super appreciated, so thanks again!

This will be in the next release.  There'll be an automated message here when it gets published.  I'll also tag this PR when I add the test for the new code path.

:tada::pray:

**Comment by gadicc** [comment]
@glaucoheitor, quoting from above commit:

```js
describe("historical", () => {
  // ...

  it("throws if period{1,2} gets an invalid string for new Date()", async () => {
    await expect(yf.historical("TSLA", { period1: "invalid" })).rejects.toThrow(
      /invalid date provided/
    );

    await expect(
      yf.historical("TSLA", { period1: "2022-02-022", period2: "invalid" })
    ).rejects.toThrow(/invalid date provided/);
  });

  //...
});
```

So, we:

1. **describe** a *test suite* called "historical".
2. have a test to check if **it** (historical) "throws if period{1,2} is invalid"
3. **expect** the yf.historical() command to **reject** (since it's a promise) and **toThrow** (an error), which we can match with a regexp (against the error message)

Hope that helps :)  If you want to familiarize yourself further, you pull in the latest devel (with the tests above), run `yarn test --watch src/modules/historical.spec.ts` and see what happens when you play with the test a little bit (you could e.g. give a *valid* string so the command won't throw, which will cause the test (which is expecting it to throw) to fail (because now it's not throwing).  Hope that makes sense!

**Pull request merged**
